searx/webapp.py

```diff
@@ -866,6 +888,42 @@ def autocompleter():
     return Response(suggestions, mimetype=mimetype)
 
 
+@app.route('/favicon', methods=['GET'])
+def favicon_proxy():
+    """Return proxied favicon results"""
+    url = request.args.get('q')
+
+    # malformed request
+    if not url:
+        return '', 400
+
+    # malformed request / does not have authorisation
+    if not is_hmac_of(settings['server']['secret_key'], url.encode(), request.args.get('h', '')):
+        return '', 400
+
+    resolver = request.preferences.get_value('favicon_resolver')
+
+    # check if the favicon resolver is valid
+    if not resolver or resolver not in favicon_backends:
+        return '', 400
+
+    # parse query
+    raw_text_query = RawTextQuery(url, [])
+
+    resp = search_favicon(resolver, raw_text_query)
+
+    # return 404 if the favicon is not found
+    if not resp:
+        theme = request.preferences.get_value("theme")
+        # return favicon from /static/themes/simple/img/empty_favicon.svg
+        # we can't rely on an onerror event in the img tag to display a default favicon as this violates the CSP.
+        # using redirect to save network bandwidth (user will have this location cached).
+        return redirect(url_for('static', filename='themes/' + theme + '/img/empty_favicon.svg'))
+
+    # will always return a PNG image
```

Comment:
OK, here is my first question: I don't think we should assume that all favicons are PNGs .. when I try https://f1.allesedv.com/32/google.com for example, I see that it is an `image/x-icon` mime-type ... or do you see it differently than I do?

Reply:
Yup - it's an x-icon on allesedv. It still works so it slipped by me. Maybe we can implement a way of just copying the media type of the backend.